### PR TITLE
Separate all PhantomJS-related things in distinct projects.

### PR DESCRIPTION
**Pull request by sjrd**
This is part of #2839.

* `PhantomJSEnv` and `RetryingComJSEnv` are sent to their own   `phantomJSEnv` sbt project.
* The settings and tasks used to construct the `ClassLoader` that loads jetty8 are extracted into a separate `AutoPlugin`, named `PhantomJSEnvPlugin`, in the sbt project `phantomJSEnvPlugin`.
* The tests in `sbt-plugin-test` do not mix PhantomJS-related tests with other tests anymore (in particular `jsDependencies`).

### CODE REVIEW

**Comment by sjrd** [comment]
CI for now.

**Comment by sjrd** [comment]
Ready to be reviewed :)

File: phantomjs-env/src/main/scala/org/scalajs/jsenv/phantomjs/RetryingComJSEnv.scala [hunk 1]
```diff
@@ -30,6 +31,10 @@ import scala.util.{Try, Failure, Success}
  *  property.
  *
  *  No retrying is performed for synchronous, or normal asynchronous runs.
+ *
+ *  Although `RetryingComJSEnv` is agnostic of the underlying JS env, and is
+ *  therefore not tight to PhantomJS, it is most often used to compensate for
```
Comment (gzm0) [inline phantomjs-env/src/main/scala/org/scalajs/jsenv/phantomjs/RetryingComJSEnv.scala]: tight -> tied?

Comment (sjrd) [inline phantomjs-env/src/main/scala/org/scalajs/jsenv/phantomjs/RetryingComJSEnv.scala]: Indeed. I make that mistake all the time.

**Comment by sjrd** [comment]
Typo fixed, and rebased for good measure. I'll assume this doesn't need a renewed approval ;)

**Comment by gzm0** [comment]
Nope, good to merge

**Pull request merged**